platform-hub-api/app/controllers/kubernetes/clusters_controller.rb

```diff
@@ -1,52 +1,42 @@
 class Kubernetes::ClustersController < ApiJsonController
 
-  before_action :load_kubernetes_clusters_hash_record
-
   # GET /kubernetes/clusters
   def index
     authorize! :read, :kubernetes_clusters
-    clusters = @kubernetes_clusters.data.map do |c|
-      c.with_indifferent_access.slice(:id, :description)
-    end.sort_by! do |c|
-      c[:id]
-    end
-    render json: clusters
+    clusters = Kubernetes::ClusterService.all_clusters.sort_by(&:name)
+    render json: clusters, each_serializer: KubernetesClusterSerializer
   end
 
-  # PATCH/PUT /kubernetes/clusters/:id
+  # PATCH/PUT /kubernetes/clusters/:name
   def create_or_update
     authorize! :manage, :kubernetes_clusters
 
     data = cluster_params.to_h
-    data[:id] = params[:id]  # ID in URL takes precedence
+    data[:name] = params[:name]  # Cluster name in URL takes precedence
 
     Kubernetes::ClusterService.create_or_update data
 
     AuditService.log(
       context: audit_context,
       action: 'update_kubernetes_cluster',
-      data: { id: params[:id] },
-      comment: "Kubernetes cluster '#{params[:id]}' created or updated by #{current_user.email}"
+      data: { name: data[:name] },
```

Comment:
What about old audit records? This change means it'll be harder to map old and new records after. Maybe that's okay though, given we're aiming to make a few breaking changes with this work – what do you think?

Reply:
Having said that, we don't need this anymore since we can just set the name to be the descriptor for the `auditable` (and thus the Audit record) will contain the name in a separate column.